x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/BlockStreamInput.java

```diff
@@ -21,4 +44,27 @@ public BlockStreamInput(StreamInput delegate, BlockFactory blockFactory) {
     BlockFactory blockFactory() {
         return blockFactory;
     }
+
+    /**
+     * Reads the length for a {@link ConstantNullBlock} from the stream and
+     * returns a {@link ConstantNullBlock} with that size. Subsequent calls
+     * to this method that return a block with the same length will return
+     * the same block.
+     */
+    Block readConstantNullBlock() throws IOException {
+        int positions = readVInt();
+        if (lastConstantNullBlock == null) {
```

Comment:
```java
if (lastConstantNullBlock == null || lastConstantNullBlock.getPositionCount() != positions || lastConstantNullBlock.tryIncRef() == false) {
    lastConstantNullBlock = blockFactory.newConstantNullBlock(positions);
}
return lastConstantNullBlock;
```

If we don't retain the reference in this stream, and use `tryIncRef()`, then we could avoid managing the life cycle of this input stream. WDYT? However, I am okay with this approach too.

Reply:
Spent 20 minutes trying to decide if it was ok to have a reference that may be invalid. I hadn't noticed `tryIncRef`, but that does make sense. Let me think some more! I do feel like "this is a ref, let's count it". But it is a funny ref. And tryIncRef would do the right stuff.

Reply:
I think I'll keep it the way I have it. We can flip it later if we feel the need.

Reply:
I am okay with that too.